// dc-react/components/BubbleChart.jsx
import React, { Component, PropTypes } from 'react';
import dc from 'dc';
import { Base } from './Base';

class BubbleChart extends Component {
  //updated by WhoIN Lee, 'elasticX', 'elasticY', 'xAxisLabel', 'yAxisLabel', and 'colors' added
  static propTypes = {
    renderHorizontalGridLines: PropTypes.bool,
    renderVerticalGridLines: PropTypes.bool,
    xAxisLabel: PropTypes.string,
    yAxisLabel: PropTypes.string,
    xAxis: PropTypes.func,
    yAxis: PropTypes.func,
    colorAccessor: PropTypes.func,
    keyAccessor: PropTypes.func,
    valueAccessor: PropTypes.func,
    radiusValueAccessor: PropTypes.func,
    x: PropTypes.func,
    y: PropTypes.func,
    r: PropTypes.func,
    colors: PropTypes.func,
    colorDomain: PropTypes.array
  };

  loadChart = (container) => {
    const chart = dc.bubbleChart(container);
    const helper = this.props.chartHelper(this, chart);
    helper.setProperties('renderHorizontalGridLines', 'renderVerticalGridLines', 'xAxisLabel', 'yAxisLabel', 'colorAccessor', 'keyAccessor',
                         'valueAccessor', 'radiusValueAccessor', 'x', 'y', 'r', 'colors'
                         'colorDomain');
    if (this.props.xAxis) {
      this.props.xAxis(chart.xAxis());
    }
    if (this.props.yAxis) {
      this.props.yAxis(chart.yAxis());
    }    
    chart.render();
  };

  render() {
    return (
      <div className={this.props.className} ref={this.loadChart} />
    );
  }
}

export default Base(BubbleChart);
